### PR TITLE
Fix anchor formatting for basic and advanced configs

### PR DESCRIPTION
**Pull request by karenzone**
Replaced hardcoded anchors for Basic and Advanced config headings with correct format. (Hardcoded format broke the versioned plugin doc build.)

Added anchor for "Config models" heading to support linking from Azure Module topic.

### CODE REVIEW

File: docs/index.asciidoc [hunk 1]
```diff
@@ -94,13 +94,14 @@ sure that at least one of these options is different per Event Hub:
 ** storage_container (defaults to Event Hub name if not defined)
 ** consumer_group
 
+[id="plugins-{type}s-{plugin}-eh_config_models"]
```
Comment (jsvd) [inline docs/index.asciidoc]: won't this still clash with the other versions of this plugin's documentation? I'm guessing we'll have to also include the version in the id?

Comment (karenzone) [reply docs/index.asciidoc]: Other examples using that format appeared to work, but what you're saying makes sense.  @dedemorton, will you help me with this, please?

Comment (dedemorton) [reply docs/index.asciidoc]: @jsvd Here's the line: https://github.com/elastic/docs-tools/blob/master/versioned_plugins.rb#L238

**Comment by jsvd** [comment]
Ugh, this is why I'm starting vacations on Thursday, thanks Dede <3

File: docs/index.asciidoc (hunk 1 above)
Comment (jsvd) [reply docs/index.asciidoc]: Ugh, this is why I'm starting vacations on Thursday, thanks Dede <3

**Comment by dedemorton** [comment]
@jsvd We should probably document the requirements somewhere...for plugin docs to follow the format that the script expects. :-P

**Pull request closed without merging**